File: src/main/java/no/nav/fo/veilarbregistrering/registrering/bruker/resources/RegistreringApi.java
package no.nav.fo.veilarbregistrering.registrering.bruker.resources;

import io.swagger.v3.oas.annotations.tags.Tag;
import io.swagger.v3.oas.annotations.Operation;
import no.nav.fo.veilarbregistrering.registrering.bruker.OrdinaerBrukerRegistrering;
import no.nav.fo.veilarbregistrering.registrering.bruker.SykmeldtRegistrering;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;

@Tag(name = "RegistreringResource")
public interface RegistreringApi {

    @Operation(summary = "Henter oppfølgingsinformasjon om arbeidssøker.")
    StartRegistreringStatusDto hentStartRegistreringStatus();

    @Operation(summary = "Starter nyregistrering av arbeidssøker.")
    OrdinaerBrukerRegistrering registrerBruker(OrdinaerBrukerRegistrering ordinaerBrukerRegistrering);

    @Operation(summary = "Henter siste registrering av bruker.")
    ResponseEntity<BrukerRegistreringWrapper> hentRegistrering();

    @GetMapping("/registrering")
    ResponseEntity<BrukerRegistreringWrapper> hentPaabegyntRegistrering();

    @Operation(summary = "Starter reaktivering av arbeidssøker.")
    void reaktivering();

    @Operation(summary = "Starter nyregistrering av sykmeldt med arbeidsgiver.")
    void registrerSykmeldt(SykmeldtRegistrering sykmeldtRegistrering);
}

File: src/main/kotlin/no/nav/fo/veilarbregistrering/registrering/bruker/resources/RegistreringApi.kt
package no.nav.fo.veilarbregistrering.registrering.bruker.resources

import io.swagger.v3.oas.annotations.Operation
import io.swagger.v3.oas.annotations.tags.Tag
import org.springframework.http.ResponseEntity

@Tag(name = "RegistreringResource")
interface RegistreringApi {

    @Operation(summary = "Henter oppfølgingsinformasjon om arbeidssøker.")
    fun hentStartRegistreringStatus(consumerId: String): StartRegistreringStatusDto

    @Operation(summary = "Henter siste registrering av bruker.")
    fun hentRegistrering(): ResponseEntity<BrukerRegistreringWrapper>

}